Build failure with mtl-2.3

```
src/HiFileParser.hs:565:5: error:
    • Variable not in scope:
        when
          :: Bool
             -> StateT IfaceGetState G.Get () -> StateT IfaceGetState G.Get a0
    • Perhaps you want to add ‘when’ to one of these import lists:
        ‘Control.Monad’ (src/HiFileParser.hs:21:1-72)
        ‘RIO’ (src/HiFileParser.hs:39:1-70)
    |
565 |     when (version >= "9001") $ void getPtr
    |     ^^^^
```

--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -1,5 +1,9 @@
 # Changelog for hi-file-parser
 
+## Unreleased
+
+Allow dependency on `mtl` >= 2.3.
+
 ## 0.1.2.0
 
 Add support for GHC 8.10 and 9.0 [#2](https://github.com/commercialhaskell/hi-file-parser/pull/2)

--- a/src/HiFileParser.hs
+++ b/src/HiFileParser.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP                        #-}
 {-# LANGUAGE DeriveGeneric              #-}
 {-# LANGUAGE DerivingStrategies         #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
@@ -18,7 +19,7 @@ module HiFileParser
 
 {- HLINT ignore "Reduce duplication" -}
 
-import           Control.Monad                 (replicateM, replicateM_)
+import           Control.Monad                 (replicateM, replicateM_, when)
 import           Data.Binary                   (Word64,Word32,Word8)
 import qualified Data.Binary.Get as G          (Get, Decoder (..), bytesRead,
                                                 getByteString, getInt64be,
@@ -31,7 +32,9 @@ import           Data.Char                     (chr)
 import           Data.Functor                  (void, ($>))
 import           Data.List                     (find)
 import           Data.Maybe                    (catMaybes)
+#if !MIN_VERSION_base(4,11,0)
 import           Data.Semigroup                ((<>))
+#endif
 import qualified Data.Vector                   as V
 import           GHC.IO.IOMode                 (IOMode (..))
 import           Numeric                       (showHex)
@@ -41,7 +44,8 @@ import           System.IO                     (withBinaryFile)
 import           Data.Bits                     (FiniteBits(..),testBit,
                                                 unsafeShiftL,(.|.),clearBit,
                                                 complement)
-import           Control.Monad.State
+import           Control.Monad.State           (StateT, evalStateT, get, gets,
+                                                lift, modify)
 import qualified Debug.Trace
 
 newtype IfaceGetState = IfaceGetState
@@ -598,7 +602,7 @@ fromFile fp = withBinaryFile fp ReadMode go
           feed (G.Partial k) = do
             chunk <- hGetSome h defaultChunkSize
             feed $ k $ if B.null chunk then Nothing else Just chunk
-      in feed $ runGetIncremental getInterface 
+      in feed $ runGetIncremental getInterface
 
 
 getULEB128 :: forall a. (Integral a, FiniteBits a) => Get a

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,1 +1,1 @@
-resolver: nightly-2019-07-15
+resolver: lts-19.18

--- a/stack.yaml.lock
+++ b/stack.yaml.lock
@@ -6,7 +6,7 @@
 packages: []
 snapshots:
 - completed:
-    size: 519764
-    url: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/nightly/2019/7/15.yaml
-    sha256: 3f93a8081b0d3e6ee2b463886b1cc9f60403fae8a2eaeab4701b6e6150dfc98d
-  original: nightly-2019-07-15
+    sha256: 65b9809265860e085b4f61d4eb00d5d73e41190693620385a69cc9d9df7a901d
+    size: 619164
+    url: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/19/18.yaml
+  original: lts-19.18

